lib/dynflow/testing/in_thread_executor.rb

```diff
@@ -34,7 +34,7 @@ def event(execution_plan_id, step_id, event, future = Concurrent.future)
       end
 
       def clock_tick
-        @world.clock.progress
+        @world.clock.progress_all
```

Comment:
One suggestion: the `progress_all` method could take an array of events that should be skipped (matching the `what` attribute of the event). We can state here the events that we don't care about with the InThreadExecutor (such as `:periodic_check_inbox` from https://github.com/Dynflow/dynflow/blob/master/lib/dynflow/connectors/database.rb#L77, that was causing the troubles in your case.